app/controllers/DashboardController.scala

```
@@ -102,23 +107,20 @@ class DashboardController @Inject() (
         maybeUserAnswers.getOrElse(UserAnswers(request.userId))
         for {
           obligationsResponse <- osService.handleData(plrReference, LocalDate.now().minusYears(SubmissionAccountingPeriods), LocalDate.now())
-          rawFinancialData <-
-            financialDataService
-              .retrieveFinancialData(plrReference, LocalDate.now().minusYears(SubmissionAccountingPeriods), LocalDate.now())
-              .map(Some(_))
-              .recover { case _ =>
-                None
-              }
```

My view is that we should avoid silently swallowing errors like this. It also makes the downstream logic difficult if not impossible, because we need to render the screen without all the data we require. Treating everything as sequences or optional risks us presenting an inaccurate view to the user.